flexdll_initer.c

```diff
@@ -24,8 +24,16 @@ extern int reloctbl;
 
 static int flexdll_init() {
   func *sym = 0;
-  char *s = getenv("FLEXDLL_RELOCATE");
-  if (!s) { fprintf(stderr, "Cannot find FLEXDLL_RELOCATE\n"); return FALSE; }
+  char *s, buf[256];
+  DWORD len = GetEnvironmentVariable("FLEXDLL_RELOCATE", buf, sizeof(buf));
+  if (len > 0 && len <= sizeof(buf)) {
+    s = buf;
+  }
+  else {
+    /* fallback to the legacy method */
+    s = getenv("FLEXDLL_RELOCATE");
+    if (!s) { fprintf(stderr, "Cannot find FLEXDLL_RELOCATE\n"); return FALSE; }
+  }
```

Comment:
Being somewhat pedantic, shouldn't this be the other way around - you start off using `getenv` and if that fails because `FLEXDLL_RELOCATE` is not set then you use the new behaviour and read `GetEnvironmentVariable`?

Reply:
I don't see much difference but if you think it's important I can do it your way.

Reply:
Just a highly pedantic "do it the old way and only if that fails do it the new way"! See my other comment about the `putenv` stuff - I'm slightly confused by it and curious to see the failure. The issue with putenv/getenv will not be that putenv doesn't call `SetEnvironmentVariable`, it's that getenv cached `GetEnvironmentVariable` (or rather `GetEnvironmentStrings` or whatever it's called)

Reply:
I was trying to work out if a program loading a flexdll DLL which also loads a flexdll DLL could fail - but that shouldn't, because it's always the same flexdll_relocate you want!